packages/scaffold-ui/src/views/w3m-connecting-wc-view/index.ts

```diff
@@ -72,7 +74,7 @@ export class W3mConnectingWcView extends LitElement {
           OptionsController.state.hasMultipleAddresses
         ) {
           RouterController.push('SelectAddresses')
-        } else {
+        } else if (!this.isSiweEnabled) {
           ModalController.close()
         }
```

Comment:
When we did `RouterController.replace('ConnectingSiwe')`, right after the modal would be closed. I did a small check just to make sure if siwe is not enabled then we go ahead and close the modal.